makeflow/src/makeflow.c

```diff
@@ -587,6 +593,54 @@ docker run --rm -m 1g -v $curr_dir:$default_dir -w $default_dir %s \"$@\"\n", im
 
 	makeflow_wrapper_add_input_file(CONTAINER_SH);
 }
+static int makeflow_is_local_node(struct dag_node *n)
+{
+	return (batch_queue_type == BATCH_QUEUE_TYPE_LOCAL || (n->local_job && local_queue));
+}
+static int makeflow_can_alloc_local(struct dag_node *n)
+{
+	int mem_ok = 0;
+	int disk_ok = 0;
+	int cores_ok = 0;
+	mem_ok = (loc_info.local_mem == -1 || !n->resources || n->resources->resident_memory >= loc_info.local_mem);
+	disk_ok = (loc_info.local_disk == -1 || !n->resources || n->resources->swap_memory >= loc_info.local_disk);
```

Comment:
Instead of n->resource->swap_memory you will want to use n->resource->workdir_footprint as this is the value used to describe disk space in rmsummary.